how-to: update binlog deploy info (#1879)
pingcap/docs-cn: dev/how-to/deploy/tidb-binlog.md

        > **注意：**
        >
        > 配置文件名命名规则为 `别名_drainer.toml`，否则部署时无法找到自定义配置文件。

>         > v3.0.2 及以后版本 ansible 不再支持 binlog kafka 模式，请勿将 drainer.toml 更名为 drainer_mysql_drainer-cluster.toml

Reviewer: 这文档就从来没讲过 *_drainer-cluster.toml 的

如果是中间某几个版本 要用 别名_drainer-cluster.toml 就直接讲哪些版本 要 这样好了（3.0 的文档)
dev 可以不写了

binlog kafka 模式也直接不讲了吧, 没对外使用过

        # Valid values are "mysql", "file", "kafka", "flash".
        db-type = "file"

>         # Uncomment this if you want to use file as db-type.

Reviewer 2: ```suggestion
        # Uncomment this if you want to use "file" as "db-type".
```

        -c int
            同步下游的并发数，该值设置越高同步的吞吐性能越好 (default 1)
        -cache-binlog-count int
            缓存中的 binlog 数目限制（默认 512）

>             如果上游的单个 binlog 较大 drainer 出现 OOM 时可尝试调小该值减少内存使用

Reviewer 2: ```suggestion
            如果上游的单个 binlog 较大导致 Drainer 出现 OOM 时，可尝试调小该值减少内存使用
```

> - 通过给 TiDB 增加启动参数 `enable-binlog` 来开启 binlog 服务。尽量保证同一集群的所有 TiDB 都开启了 binlog 服务，否则在同步数据时可能会导致上下游数据不一致。如果要临时运行一个不开启 binlog 服务的 TiDB 实例，需要在 TiDB 的配置文件中设置 `run_ddl= false`。
> - Drainer 不支持对 ignore schemas（在过滤列表中的 schemas）的 table 进行 rename DDL 操作。
> - 在已有的 TiDB 集群中启动 Drainer，一般需要全量备份并且获取 savepoint，然后导入全量备份，最后启动 Drainer 从 savepoint 开始同步增量数据。

> > - 下游使用 MySQL 或 TiDB 时应当保证上下游数据库的 sql_mode 具有一致性；同时如果存在上游 TiDB 支持但下游 MySQL 不支持的 DDL 语句时（例如`create table binlog_split_region(a int, b int) pre_split_regions=3;`） drainer 也会同步失败，此时可以在 drainer 配置中跳过该事务同时在下游手动执行兼容的语句。

Author: 需要放这个[TOOL-1596](https://internal.pingcap.net/jira/browse/TOOL-1596)里面提到的 Row match 吗？还是说直接什么例子不放只留这句话

> > - 下游使用 MySQL 或 TiDB 时应当保证上下游数据库的 sql_mode 具有一致性；同时如果存在上游 TiDB 支持但下游 MySQL 不支持的 DDL 语句时（例如`create table binlog_split_region(a int, b int) pre_split_regions=3;`） drainer 也会同步失败，此时可以在 drainer 配置中跳过该事务同时在下游手动执行兼容的语句。

Reviewer: 提下吧

> - 通过给 TiDB 增加启动参数 `enable-binlog` 来开启 binlog 服务。尽量保证同一集群的所有 TiDB 都开启了 binlog 服务，否则在同步数据时可能会导致上下游数据不一致。如果要临时运行一个不开启 binlog 服务的 TiDB 实例，需要在 TiDB 的配置文件中设置 `run_ddl= false`。
> - Drainer 不支持对 ignore schemas（在过滤列表中的 schemas）的 table 进行 rename DDL 操作。
> - 在已有的 TiDB 集群中启动 Drainer，一般需要全量备份并且获取 savepoint，然后导入全量备份，最后启动 Drainer 从 savepoint 开始同步增量数据。

> > - 下游使用 MySQL 或 TiDB 时应当保证上下游数据库的 sql_mode 具有一致性；同时如果存在上游 TiDB 能运行但下游 MySQL 不支持的 DDL 语句时（例如下游 MySQL 使用 InnoDB 引擎时同步语句`CREATE TABLE t1(a INT) ROW_FORMAT=FIXED;`） drainer 也会同步失败，此时可以在 drainer 配置中跳过该事务同时在下游手动执行兼容的语句。

Reviewer 3: 怎么在 drainer 配置中跳过该事务，有操作文档吗，可以 link 过来，要不用户还得自己去找相关的操作手册。 其他的用户文档也注意一下，对用户友好点

Author: 找了一下似乎没找到类似的操作文档。我补一个到 https://pingcap.com/docs-cn/v3.0/how-to/maintain/tidb-binlog/ 的 FAQ 下面然后再 link 过来吧

Author: 我现在是加到 FAQ 里然后引了个 link 过去。需要删了后然后 link 到配置文件吗？

Reviewer: 现在这样也可以吧 不过好像 maintain 讲的 FAQ 都是状态管理的

> - 通过给 TiDB 增加启动参数 `enable-binlog` 来开启 binlog 服务。尽量保证同一集群的所有 TiDB 都开启了 binlog 服务，否则在同步数据时可能会导致上下游数据不一致。如果要临时运行一个不开启 binlog 服务的 TiDB 实例，需要在 TiDB 的配置文件中设置 `run_ddl= false`。
> - Drainer 不支持对 ignore schemas（在过滤列表中的 schemas）的 table 进行 rename DDL 操作。
> - 在已有的 TiDB 集群中启动 Drainer，一般需要全量备份并且获取 savepoint，然后导入全量备份，最后启动 Drainer 从 savepoint 开始同步增量数据。

> > - 下游使用 MySQL 或 TiDB 时应当保证上下游数据库的 sql_mode 具有一致性，即下游数据库同步每条 sql 语句时的 sql_mode 应当与上游数据库执行该条 sql 语句时的 sql_mode 保持一致。可以在上下游分别执行 `select @@sql_mode;` 进行查询和比对。

Reviewer 2: ```suggestion
> - 下游使用 MySQL 或 TiDB 时应当保证上下游数据库的 sql_mode 具有一致性，即下游数据库同步每条 SQL 语句时的 sql_mode 应当与上游数据库执行该条 SQL 语句时的 sql_mode 保持一致。可以在上下游分别执行 `select @@sql_mode;` 进行查询和比对。
```

> > - 如果存在上游 TiDB 能运行但下游 MySQL 不支持的 DDL 语句时（例如下游 MySQL 使用 InnoDB 引擎时同步语句`CREATE TABLE t1(a INT) ROW_FORMAT=FIXED;`） drainer 也会同步失败，此时可以在 drainer 配置中跳过该事务同时在下游手动执行兼容的语句，详见[跳过事务](/dev/how-to/maintain/tidb-binlog.md#同步时出现上游数据库支持但是下游数据库执行会出错的-DDL应该怎么办)。

Reviewer 2: ```suggestion
> - 如果存在上游 TiDB 能运行但下游 MySQL 不支持的 DDL 语句时（例如下游 MySQL 使用 InnoDB 引擎时同步语句 `CREATE TABLE t1(a INT) ROW_FORMAT=FIXED;`），Drainer 也会同步失败，此时可以在 Drainer 配置中跳过该事务，同时在下游手动执行兼容的语句，详见[跳过事务](/dev/how-to/maintain/tidb-binlog.md#同步时出现上游数据库支持但是下游数据库执行会出错的-DDL应该怎么办)。
```
